src/include/sof/audio/module_adapter/module/generic.h

```diff
@@ -82,6 +82,15 @@ enum module_state {
 	MODULE_PROCESSING, /**< Module is processing samples now. */
 };
 
+/**
+ * \enum module_retcode
+ * \brief Module-specific return codes - 0 is preserved for function success and
+ * all negative values are preserved for errors.
+ */
+enum module_retcode {
+	MODULE_PREPARE_REQUESTED = 1, /**< Module requests prepare() call after reset. */
```

Comment:
@johnylin76 I want to remove the MODULE_* states and replace them with the existing COMP_STATE_* in the near future.

Reply:
@ranj063 do you mean the alignment for module state and module_adapter state (combine them into one)? It sounds fine to me as well.
Could you also take a look of #6166 and provide comments? Would it be covered by the state alignment?
Thanks

Reply:
@johnylin76 I think the fix in #6184 is probably the right thing to do with the changes to remove the prepare() call inside the cadence codec's reset and addition of memory free in all other modules' reset callbacks.  This should handle the issue with volume in #6034 and also fix the alsa conformance test failures that you are seeing. Once this is done, I will modify the module adapter to remove the MODULE_IDLE/INITIALIZED states and use the comp states. Let me open a new PR Hope that is OK